src/components/TaskList/index.jsx

```diff
@@ -1,38 +1,54 @@
 import React, { useEffect, useState } from "react";
 import PropTypes from "prop-types";
-import { useDispatch } from "react-redux";
+import { useDispatch, useSelector } from "react-redux";
 import { paginationLimitUpdate } from "store/actions";
+import Task from "components/TaskCard/Task";
 import {
   FILTER_STATE_ALL,
   FILTER_STATE_COMPLETE,
   FILTER_STATE_INCOMPLETE,
   PAGINATION_LIMIT,
 } from "utils/constant";
-import Task from "components/TaskCard/Task";
 
 const TaskList = ({ tasks, limit, filter, setTaskLength, isCardCreated }) => {
   const [filteredTasks, setFilteredTasks] = useState([]);
+  const searchText = useSelector((state) => state.searchText);
+
   const dispatch = useDispatch();
   function getCompletedTasks() {
-    return tasks.filter((todo) => todo.completed == true);
+    return tasks
+      .filter((todo) => todo.completed == true)
+      .filter((task) =>
+        task.task.toLowerCase().includes(searchText.toLowerCase())
+      );
   }
 
   function getIncompletedTasks() {
-    return tasks.filter((todo) => todo.completed !== true);
+    return tasks
+      .filter((todo) => todo.completed !== true)
+      .filter((task) =>
+        task.task.toLowerCase().includes(searchText.toLowerCase())
+      );
   }
+  function getAllTasks() {
+    return tasks.filter((task) =>
+      task.task.toLowerCase().includes(searchText.toLowerCase())
+    );
+  }
+
   useEffect(() => {
     switch (filter) {
       case FILTER_STATE_COMPLETE:
-        setFilteredTasks(getCompletedTasks());
+        setFilteredTasks(getCompletedTasks);
```

Comment:
Are we calling `getCompletedTasks` function or passing function reference here ?